src/django/api/reports.py

```diff
@@ -4,15 +4,37 @@
 from urllib.parse import quote
 
 from django.db import connection
-
+from api.models import HistoricalFacility
 
 _root = os.path.abspath(os.path.dirname(__file__))
 
 
+def monthly_promoted_name_and_address():
+    data = dict()
+    historical_facilities = HistoricalFacility.objects \
+        .exclude(created_from_id__isnull=True) \
+        .order_by('id', 'history_date') \
+        .values_list('id', 'history_date', 'created_from_id')
+    for i, h in enumerate(historical_facilities):
+            if i != 0:
+                prev = historical_facilities[i - 1]
+                if prev[0] == h[0] and h[2] != prev[2]:
+                    month = h[1].strftime('%Y-%m')
+                    data[month] = data.get(month, 0) + 1
+    return [['month', 'facilities promoted'],
+            sorted(data.items(), key=lambda x: x[0], reverse=True)]
```

Comment:
I'm glad this optimization worked out and the logic is basically the same as it was.

I am seeing the result sorted in descending order. Our other reports are in ascending order, with the newest row at the bottom, so we should consider changing this report to match the rest.

<img width="884" alt="Screen Shot 2021-06-15 at 10 21 45 AM" src="https://user-images.githubusercontent.com/17363/122096941-f2c14000-cdc3-11eb-9eac-ec6642fab29e.png">

Reply:
Thanks! I've updated the sort order.